### PR TITLE
Pinak dev

### PR DESCRIPTION
**Pull request by Pinak08**
comment data file is completed.
Troy you need to add two array fields of upvotedComments and downvotedComments in your user data.

### CODE REVIEW

File: data/comments.js [hunk 1]
```diff
@@ -0,0 +1,907 @@
...
+      }
+    );
+    if (!updateInfo4.matchedCount && !updateInfo4.modifiedCount)
+      throw 'Removing Downvote failed';
+  }
+  return { msg: "Remove Downvote Successful!" }
+}
+//---------------------------------------------------------------------------------------------------------
+
+async function mostUpvoted(reviewId) {
```
Comment (troytomasch) [inline data/comments.js:835]: What is the purpose of the mostUpvoted and mostDownvoted functions?

**Review by troytomasch — CHANGES_REQUESTED** [review]
Need to remove a few things and I just had a few questions on some of the functions

File: data/comments.js (hunk 1 above)
Comment (Pinak08) [reply data/comments.js:835]: Most upvoted function will sort all comments in order of most upvoted to most downvoted and viceversa for most downvoted.

This is used for filtering comments.

**Comment by rgundam-stevens** [comment]
________________________________________

File: data/comments.js [hunk 2]
```diff
@@ -0,0 +1,907 @@
...
+  /* Updating the review text */
+
+  const updatedInfo = await commentCollection.updateOne(
+    { _id: ObjectId(id) },
+    { $set: updatedComment }
+  );
+  if (updatedInfo.modifiedCount === 0) {
+    throw 'could not update comment text successfully';
+  }
+  const comment = await this.getCommentById(id);
```
[the remark this answers is not in the document]
Comment (Pinak08) [reply data/comments.js:265]: Will do the needful.

File: data/comments.js [hunk 3]
```diff
@@ -0,0 +1,907 @@
...
+
+  const reviewThatGotPosted = await review.getReviewById(reviewId);
+  if (!reviewThatGotPosted) throw 'Review not found';
+
+
+  let comment = null;
+  try {
+    comment = await this.getCommentById((id));
+  } catch (e) {
+    console.log(e);
```
[the remark this answers is not in the document]
Comment (Pinak08) [reply data/comments.js:161]: will implement the changes.

File: data/comments.js [hunk 4]
```diff
@@ -0,0 +1,907 @@
...
+    throw 'review Id should be valid ObjectId';
+  }
+
+  try {
+    ObjectId(id);
+  } catch (error) {
+    throw 'comment ID Id should be valid ObjectId';
+  }
+  
+  const reviewCollection = await reviews();
```
[the remark this answers is not in the document]
Comment (Pinak08) [reply data/comments.js:123]: will comment it out.

File: data/comments.js [hunk 5]
```diff
@@ -0,0 +1,907 @@
...
+  }
+  
+  try {
+    ObjectId(reviewId);
+  } 
+  catch (error) {
+    throw 'review Id should be valid ObjectId';
+  }
+
+  const reviewCollection = await reviews();
```
[the remark this answers is not in the document]
Comment (Pinak08) [reply data/comments.js:36]: Done!

**Review by troytomasch — APPROVED** [review]

**Pull request merged**